File: Java/012.java
public class Problem012 {

    public static int divisors(long n) {
        int count = 0;
        for (int i = 1; i * i <= n; i++) {
            if (n % i == 0) {
                if (n / i == i) {
                    count++;
                } else {
                    count += 2;
                }
            }
        }
        return count;
    }

    public static void main(String[] args) {
        long n = 1;
        while (true) {
            long triangleNumber = (n * (n + 1)) / 2;
            if (divisors(triangleNumber) >= 500) {
                System.out.println(triangleNumber);
                break;
            }
            n++;
        }
    }
}

File: Rust/012.rs
fn count_divisors(n: u64) -> u32 {
    let mut count = 0;
    let limit = (n as f64).sqrt() as u64;

    for i in 1..=limit {
        if n % i == 0 {
            if n / i == i {
                count += 1;
            } else {
                count += 2;
            }
        }
    }
    count
}

fn main() {
    let mut n: u64 = 1;
    loop {
        let triangle_number = n * (n + 1) / 2;
        if count_divisors(triangle_number) >= 500 {
            println!("{}", triangle_number);
            break;
        }
        n += 1;
    }
}
